nominal/nominal.py

```diff
@@ -1,628 +1,316 @@
-import requests
-from dateutil import parser
-from jsondiff import diff
-from rich import print
-
-from ._api.combined.ingest_api import (
-    AbsoluteTimestamp,
-    IngestService,
-    IngestSource,
-    RelativeTimestamp,
-    S3IngestSource,
-    TimestampMetadata,
-    Iso8601Timestamp,
-    TimestampType,
-    TriggerIngest,
+from threading import Thread
+from typing import TYPE_CHECKING, BinaryIO, Mapping, Sequence
+
+import dateutil.parser
+
+from .exceptions import NominalError
+from .sdk import Attachment, Dataset, NominalClient, Run
+from ._utils import (
+    FileType,
+    FileTypes,
+    IntegralNanosecondsUTC,
+    TimestampColumnType,
+    _datetime_to_integral_nanoseconds,
+    reader_writer,
 )
-from ._utils import PayloadFactory, create_service, default_filename
-from .cloud import ENDPOINTS, _auth_help_blurb, get_app_base_url, get_base_url
 
+if TYPE_CHECKING:
+    import pandas as pd
+    import polars as pl
 
-class Dataset(pl.DataFrame):
-    """
-    Dataset inherits from Polars DataFrame for its rich display, ingestion, and wrangling capabilities.
-
-    Parameters
-    ----------
-    data : various, optional
-        The input data for the dataset. This can be in any format supported by Polars DataFrame.
-    filename : str, optional
-        The name of the dataset file. Default is None.
-    overwrite : bool, optional
-        A flag to indicate whether to overwrite an existing file during upload. Default is False.
-    properties : dict, optional
-        A dictionary of additional properties associated with the dataset. Default is an empty dictionary.
-    description : str, optional
-        A brief description of the dataset. Default is an empty string.
-
-    Attributes
-    ----------
-    s3_path : str or None
-        The S3 path where the dataset is stored after upload. Initially None.
-    filename : str
-        The name of the dataset file.
-    properties : dict
-        A dictionary of additional properties associated with the dataset.
-    description : str
-        A brief description of the dataset.
-    rid : str or None
-        The dataset's RID (Resource ID) after registration on the Nominal platform. Initially None.
-    dataset_link : str
-        A URL link to the dataset on the Nominal platform. Initially an empty string.
-
-    Methods
-    -------
-    upload(overwrite=False)
-        Uploads and registers the dataset on the Nominal platform.
+
+_default_connection: NominalClient | None = None
+
+
+def set_default_connection(base_url: str, token: str) -> None:
+    """Set the default global connection to the Nominal platform.
+
+    base_url: The URL of the Nominal API platform, e.g. "https://api.gov.nominal.io/api".
+    token: An API token to authenticate with. You can grab a client token from the Nominal sandbox, e.g.
+        at https://app.gov.nominal.io/sandbox.
     """
+    global _default_connection
+    _default_connection = NominalClient.create(base_url, token)
```

Comment:
thoughts on `NominalClient.get()` or `.default()` instead of a global?

Reply:
still need a global for the url or token either way (or data persisted to disk) so I don't think this is majorly different